nucleus/security/core/src/test/java/com/sun/enterprise/security/auth/realm/ldap/GroupSearchLoggerTest.java

```diff
@@ -0,0 +1,83 @@
+ *
+ * Copyright (c) [2018] Payara Foundation and/or its affiliates. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of either the GNU
+ * General Public License Version 2 only ("GPL") or the Common Development
+ * and Distribution License("CDDL") (collectively, the "License").  You
+ * may not use this file except in compliance with the License.  You can
+ * obtain a copy of the License at
+ * https://github.com/payara/Payara/blob/master/LICENSE.txt
+ * See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing the software, include this License Header Notice in each
+ * file and include the License file at glassfish/legal/LICENSE.txt.
+ *
+ * GPL Classpath Exception:
+ * The Payara Foundation designates this particular file as subject to the "Classpath"
+ * exception as provided by the Payara Foundation in the GPL Version 2 section of the License
+ * file that accompanied this code.
+ *
+ * Modifications:
+ * If applicable, add the following below the License Header, with the fields
+ * enclosed by brackets [] replaced by your own identifying information:
+ * "Portions Copyright [year] [name of copyright owner]"
+ *
+ * Contributor(s):
+ * If you wish your version of this file to be governed by only the CDDL or
+ * only the GPL Version 2, indicate your decision by adding "[Contributor]
+ * elects to include this software in this distribution under the [CDDL or GPL
+ * Version 2] license."  If you don't indicate a single choice of license, a
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package com.sun.enterprise.security.auth.realm.ldap;
+
+import java.util.logging.Logger;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import static org.junit.Assert.*;
+import com.sun.enterprise.security.util.LogAppender;
+import java.util.logging.Level;
+import static java.util.logging.Level.INFO;
+import static java.util.logging.Level.WARNING;
+/**
+ *
+ * @author Alan Roth
+ */
+public class GroupSearchLoggerTest{
+    LogAppender logAppender;
+    Logger logger;
+    
+    @Before
+    public void setUp(){
+        logAppender = new LogAppender();
+        logger = LDAPRealm._groupSearchLogger;
+        logger.addHandler(logAppender);
+    }
+    
+    @Test
+    public void groupSearchErrorLoggingTest() {
+        logger.log(INFO, "ldaprealm.groupsearcherror");
+        assertNotNull(logAppender.getLogs());
+        assertEquals(1, logAppender.getLogs().size());
+        assertEquals("ldaprealm.groupsearcherror", logAppender.getLogs().get(0).getMessage());
```

Comment:
This test doesn't test the code changes but properties of the logging framework which hopefully already has that covered. Looking at the change I don't think there is a meaningful test other than making sure that the `LDAPRealm` service actually uses the group logger that you wanted it to use. I think you could just do a `new LDAPRealm()`, grab the logger field from it and check its name against your expectations.